### ISSUE TITLE
When specifying config file psalm does not find any errors

### ISSUE THREAD

**Issue by simPod** [vimeo/psalm#6056]
Create psalm config using `vendor/bin/psalm --init src 1`

Running 

```sh
vendor/bin/psalm --config psalm.xml

Scanning files...
Analyzing files...

░

To whom it may concern: Psalm cannot detect unused classes, methods and properties
when analyzing individual files and folders. Run on the full project to enable
complete unused code detection.
------------------------------
No errors found!
------------------------------
```
finds 0 errors

----------

Running 

```sh
vendor/bin/psalm

Scanning files...
Analyzing files...

░░░░░░░E░░░░░░E░░░░░░░░░░░░

...

------------------------------
4 errors found
------------------------------

```

finds some errors. Why is it so? I though I could have multiple psalm configs and specify them using `--config` flag while it seems the flag disables the analysis.

**Comment by weirdan** [comment]
Does it work as expected if you run it as
```
vendor/bin/psalm --config=psalm.xml
```
? (note the equals sign)

**Comment by simPod** [comment]
Oh it does, thanks.

**Comment by weirdan** [comment]
I thought it was how `getopt()` parses that line, but it's not `getopt()`s fault:

```console
$ php -r 'var_dump(getopt("", ["config:"]));' -- --config config.xml
Command line code:1:
array(1) {
  'config' =>
  string(10) "config.xml"
}
```

**Comment by simPod** [comment]
Is there a way to run psalm with xdebug on? I tried to debug it.

**Comment by orklah** [comment]
In Psalm.php there is a specific code that disable xdebug. You can comment the "check" method call IIRC and xdebug will run.

**Comment by weirdan** [comment]
Just run it with `PSALM_ALLOW_XDEBUG` env variable set ([docs](https://psalm.dev/docs/running_psalm/plugins/authoring_plugins/#using-xdebug)):

```console
$ PSALM_ALLOW_XDEBUG=1 psalm
```

**Comment by simPod** [comment]
Even better. The issue is in `CliUtils::getPathsToCheck()`

**Comment by simPod** [comment]
I'll send fix soon.

**Comment by weirdan** [comment]
Going to keep this open as I would like to address this in a more general way.
